### ISSUE TITLE
Compile time warning in UIColor.init(rgbaValue:)

### ISSUE THREAD

**Issue by ryanmasondavies** [SwiftGen/SwiftGen#704]
This generated function can take a non-trivial amount of time to compile:

https://github.com/SwiftGen/SwiftGen/blob/c6477a6950d313c94c964c6c8ec960e3aeccb8f6/templates/colors/swift4.stencil#L49-L58

This was flagged by `-Xfrontend -warn-long-expression-type-checking=250 -Xfrontend -warn-long-function-bodies=250`. The function was taking around 300ms to compile.

I propose an equivalent implementation that is faster to compile:

```swift
// swiftlint:disable operator_usage_whitespace
internal extension {{colorAlias}} {
  convenience init(rgbaValue: UInt32) {
    let components = rgbaValue.normalizedRGBAComponents
    self.init(red: components[0], green: components[1], blue: components[2], alpha: components[3])	
  }
}
// swiftlint:enable operator_usage_whitespace

fileprivate extension UInt32 {
  var rgbaComponents: [CGFloat] {
    let shifts: [UInt32] = [
      self >> 24, // red
      self >> 16, // green
      self >> 8,  // blue
      self        // alpha
    ]
    return shifts.map {
      CGFloat($0 & 0xff)
    }
  }

  var normalizedRGBAComponents: [CGFloat] {
    rgbaComponents.map { $0 / 255.0 }
  }
}
```

Using a custom template with this implementation resolved the issue for us, but it would be great to get this or an equivalent fix made so we can use the built-in template.

**Comment by AliSoftware** [comment]
That seems like a reasonable change to make, thanks a lot for flagging this and for the suggestion!

Could you maybe create a PR to add that to the existing `swift4` and `swift5` color templates in the repo? You'd just have to:
 - make that change in the `Resources/templates/colors/*.stencil` files
 - then open the project in Xcode and run the "Templates - Generate Outputs" scheme so that it automatically update the unit tests with the new template output
 - Credit yourself in the CHANGELOG

Thanks 👍

**Comment by ryanmasondavies** [comment]
I hadn't seen #664, which also attempts to solve this problem. I've tested the changes made by that PR and found it does not resolve the compile time warning for us (at least in my test project, which is very large). As such, I'll be proceeding with making this change.

**Comment by AliSoftware** [comment]
Closed by #710
